src/SpectatorView.Native/SpectatorView.Compositor/Compositor/VideoEncoder.cpp

```diff
@@ -21,10 +23,11 @@ VideoEncoder::VideoEncoder(UINT frameWidth, UINT frameHeight, UINT frameStride,
     isRecording(false)
 {
 #if HARDWARE_ENCODE_VIDEO
-  inputFormat = MFVideoFormat_NV12;
+    inputFormat = MFVideoFormat_NV12;
 #else
-  inputFormat = MFVideoFormat_RGB32;
+    inputFormat = MFVideoFormat_RGB32;
 #endif
+    inputFormat = MFVideoFormat_RGB32;
 }
```

Comment:
Does this change mean that hardware encoding succeeds using RGB32?

Reply:
For me it *only* worked with RGB32, I guess this is because the converted texture is still reported to D3D as being RGB. While it gets the information that the input is supposed to be NV12 with a therefore smaller buffer.
MF seems to be confused and writing samples failed with MF_E_BUFFERTOOSMALL.